FlowCrypt/Functionality/Mail Provider/Threads/MessagesThreadApiClient.swift

```diff
@@ -60,7 +60,15 @@ extension GmailService: MessagesThreadApiClient {
 
     func fetchThread(identifier: String, path: String) async throws -> MessageThread {
         let query = GTLRGmailQuery_UsersThreadsGet.query(withUserId: .me, identifier: identifier)
-        query.format = kGTLRGmailFormatMetadata
+        query.format = kGTLRGmailFormatFull
```

Comment:
I think we used `metadata` here for improving performance, as fetching all messages for all threads can take too much time when there are several large threads with many messages.

Maybe there is some other way to check if thread messages contain attachments?

Reply:
Well, maybe...
But keep in mind that we also need to check if attachment is public key or not.

Reply:
We can simplify and not differentiate public keys and other attachments, if it'll improve performance

Reply:
By the way, how it's done on Android - does it fetch `full` format for threads?

Reply:
Need to check

Reply:
As far as i checked, in Android app it also fetches `full` format from Gmail API.
@DenBond7 Please correct me if I'm wrong.

Reply:
> By the way, how it's done on Android - does it fetch full format for threads?

Just in case, Android doesn't support threads yet. 

> As far as i checked, in Android app it also fetches full format from Gmail API.
@DenBond7 Please correct me if I'm wrong.

Yes, we use `full` format to make some requests. But we don't load a full message. Please look at this code

https://github.com/FlowCrypt/flowcrypt-android/blob/1f653a0b24844ceeae0dfc2f6a4c273e200ff4bf/FlowCrypt/src/main/java/com/flowcrypt/email/api/email/gmail/GmailApiHelper.kt#L115

Reply:
But gmail threads api only accept `format` & `metadataHeaders` params.

https://github.com/FlowCrypt/flowcrypt-android/blob/1f653a0b24844ceeae0dfc2f6a4c273e200ff4bf/FlowCrypt/src/main/java/com/flowcrypt/email/api/email/gmail/GmailApiHelper.kt#L787C11-L787C11

![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/77344191/afbc68b8-6048-41d8-b2a2-ebccd289b786)

Reply:
Does Android `GmailApiService` use another api or use some inner handling of fields param?

Reply:
For now, Android uses `users.messages` and doesn't use `users.threads`

Reply:
Yep, you are right.
But from gmail api documentation, `users.messsages` api also doesn't accept `fields` api.

Reply:
Probably we can use data received with `metadata` format to determine message encryption - if it's `snippet` property starts with `-----BEGIN PGP MESSAGE-----` then message is encrypted.

Reply:
> Yep, you are right. But from gmail api documentation, `users.messsages` api also doesn't accept `fields` api.

[this one ](https://developers.google.com/gmail/api/reference/rest/v1/users.messages/get) accepts

![image](https://github.com/FlowCrypt/flowcrypt-ios/assets/2863246/f0afa47b-5d37-4c28-a04a-6d6cefc06dc2)

Reply:
> Probably we can use data received with `metadata` format to determine message encryption - if it's `snippet` property starts with `-----BEGIN PGP MESSAGE-----` then message is encrypted.

to be honest, without loading the full message it's not possible to determine 100% if the message is encrypted or not. It was true at least for IMAP protocol. we have similar functionality on Android and it doesn't work well. From time to time we receive feedback from users that some message is not marked as encrypted.

`Gmail API` has `snippet`, but this info can be fetched only via `users.messages.get`. We can use [Batching Requests](https://developers.google.com/gmail/api/guides/batch) to improve the performance when we load the message list and fetch details about each message. But I'm still not sure that this will cover all cases to determine if the message is encrypted or not(need to investigate it).

Reply:
Yeah, it's more complicated on iOS as inbox screen can contain threads with a lot of messages, while Android loads a fixed number of messages. Documentation for batch requests says `Larger batch sizes are likely to trigger rate limiting. Sending batches larger than 50 requests is not recommended.`, so in some cases we won't be able to load messages for all displayed threads.

Current Ioan's implementation which uses `kGTLRGmailFormatFull` seems to be the simplest for this functionality, it'll be ok to use it, I just need to re-check how it works when user have large threads, as I thought we switched to `kGTLRGmailFormatMetadata` because of such issues.

Reply:
I tried to use it with some big messages and large threads - works well, didn't notice any lags.
Then let's leave it for now

Reply:
Ok,